### PR TITLE
Choose a valid branch with "travis compile".

### PR DESCRIPTION
**Pull request by nex3**
This defaults to the current branch, and falls back to not specifying
--branch at all and using whatever the repository default it.

Closes travis-ci/travis-ci#7286

### CODE REVIEW

File: init.rb [hunk 1]
```diff
@@ -65,10 +67,19 @@ def data
...
               :fetch_timeout => 60,
               :push_timeout => 60
+            },
+            :job => {
+              :branch => choose_branch
             }
           }
         end
 
+        def choose_branch
```
Comment (BanzaiMan) [inline init.rb]: The name is a bit off. This should be simply `branch`.

File: init.rb [hunk 2]
```diff
@@ -65,10 +67,19 @@ def data
...
+          return branch if branch
```
Comment (BanzaiMan) [inline init.rb]: Consequently, `branch` should not be an `attr_accessor`, but a wrapper to choose a proper value based on `@branch`.

File: init.rb (hunk 1 above)
Comment (nex3) [inline init.rb]: Done.

File: init.rb (hunk 2 above)
Comment (nex3) [inline init.rb]: Done.

**Comment by meatballhat** [comment]
@nex3 Sorry for yet another delay here! Same situation here given conflicts, so please holler if you're interested in picking this back up 💖

**Pull request closed without merging**